### PR TITLE
[TS migration] Improve ref types

### PR DESCRIPTION
**Pull request by jnowakow**
### Details

This PR is a follow up of the TS migration. It's aim was to figure out way of handling refs in the project as in many places the types were defined as union of React Native's components and HTML elements.

Bug caused by usage of refs was found and fixed. The `AutocompleteSuggestions` component was handling long press events by overriding `onpointerdown` handler. But the reference to dom element wasn't ready during `useEffect` hook and result was that the autocomplete was closing on mobile browsers when suggestion was pressed for longer time. It was fixed by using `onPointerDown` provided by react-native-web.

In general we should avoid using DOM elements as ref types. That's React Native Web handles references on web so there's no need to explicitly use types like `HTMLDivElement`. Exception of this rule is when we explicitly need to use functions available only in DOM and not in React Native, e.g. `getBoundingClientRect`. It's heavily used in `Tooltip` and `Popover` which are only used on web.

When adding handlers for Pointers and Mouse events the [interaction API provided by React Native Web](https://necolas.github.io/react-native-web/docs/interactions/) is preferred option. It's generally satisfied across the app.

In general if there is no need to use DOM specific methods then one should use View as ref type. This PR updates types in several places to follow this pattern. Union types like `View | HTMLDivElement` should be used only in places where DOM api is necessary. Technically components on web have type equal to `View & HTMLDivElement` because they have methods available on both platforms. But it's not true in case of native components and this is why union should be used. It enforces developer to explicitly check if metod on property exist on the reference object. While it may be a bit cumbersome it prevents runtime bugs.

Unfortunately, union types forces us to use type casting. This is because `HTMLDivElement` can't be passed to `View`. It should be done as soon as possible to prevent such types to be propagated. There are utility functions like `viewRef` defined to achieve that.

### Fixed Issues

$ https://github.com/Expensify/App/issues/39116
PROPOSAL: N/A

### Tests

1. Open chat.
2. Start typing emoji name to open autocomplete eg. :app
3. Press and hold one of available options.
4. Modal shouldn't close and keyboard shouldn't loose focus.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

Same as described in Tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/56261019/4812dd33-b278-40bf-8fb9-68276642c4e2

</details>

<details>
<summary>Android: mWeb Chrome</summary>

before: 

https://github.com/Expensify/App/assets/56261019/6b1cbb03-38ca-4d1b-8c92-2ff6c1135282

after:

https://github.com/Expensify/App/assets/56261019/f6a2491a-8d4f-405f-a7bc-b2ed06f6539d

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/56261019/d7922478-f832-48ef-8bf8-abdd8ad5f99b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/56261019/bb11f7f4-cbc5-4cd4-bca2-0e73337f74c8

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/56261019/dbd73ae9-35cb-46fd-af47-e41f25bc6773

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/56261019/c29f4751-3cc0-42e4-9f66-815dec737b7e

</details>

### CODE REVIEW

**Review by blazejkustra — CHANGES_REQUESTED** [review]
Great work! I've left some feedback, you missed some files that we should refactor. DMed with more examples on slack 🚀

File: src/hooks/useDragAndDrop/types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+type DragAndDropParams = {
+    dropZone: React.MutableRefObject<HTMLDivElement | null>;
+    onDrop?: (event: DragEvent) => void;
+    shouldAllowDrop?: boolean;
+    isDisabled?: boolean;
+    shouldAcceptDrop?: (event: DragEvent) => boolean;
+};
+
+type DragAndDropOptions = {
```
Comment (fabioh8010) [inline src/hooks/useDragAndDrop/types.ts:9]:
```suggestion
type DragAndDropResult = {
```

`DragAndDropOptions` gives the impression that is a type for supplying options to `useDragAndDrop`, which is not the case here.

**Review by fabioh8010 — COMMENTED** [review]
Did just an early review since the PR is not ready yet, but some points:

**1)** About this:

> Bug caused by usage of refs was found and fixed. The AutocompleteSuggestions component was handling long press events by overriding onpointerdown handler. But the reference to dom element wasn't ready during useEffect hook and result was that the autocomplete was closing on mobile browsers when suggestion was pressed for longer time. It was fixed by using onPointerDown provided by react-native-web.

Is it something that you discovered while working on this issue? If not, could you please link the GH issue related to the bug and add to the PR description?

**2)** I agree with your reasonings, but I think we should make this more explicit e.g adding some instructions to `TS_STYLE.md`. WDYT? cc @blazejkustra 

**3)** Please follow these instructions [here](https://github.com/Expensify/App/pull/40698#issuecomment-2069559909) to sign the CLA.

**4)** I've noticed your commits are unsigned, [all commits in the repo must be GPG-signed](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#begin-coding-your-solution-in-a-pull-request).

**Comment by blazejkustra** [comment]
> 1. Is it something that you discovered while working on this issue?

Yes, @jnowakow discovered this while working on improving ref types.

2. I think that makes sense, let's add a paragraph to TS_STYLE markdown
3. and 4. We are aware of this, thanks! @jnowakow will fix it today 😄

**Comment by jnowakow** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/hooks/useDragAndDrop/types.ts (hunk 1 above)
Comment (jnowakow) [reply src/hooks/useDragAndDrop/types.ts]: Yeah, make sense. I just moved that code but it's good idea to rename it

File: contributingGuides/TS_STYLE.md [hunk 2]
```diff
@@ -640,6 +641,21 @@ type Foo = {
   export someVariable
   ```
 
+- [1.23](#ref-types) **Ref types**: Avoid using HTML elements while declaring refs. Please use React Native components where possible. React Native Web handles the references on its own. It also extends React Native components [with Interaction API](https://necolas.github.io/react-native-web/docs/interactions/) which should be used to handle Pointers and Mouse events. If methods  Exception of this rule is when we explicitly need to use functions available only in DOM and not in React Native, e.g. `getBoundingClientRect`. Then please declare ref type as `union` of React Native component and HTML element. When passing it to React Native component cast it as soon as possible using utility methods declared in `src/types/utils`
```
Comment (blazejkustra) [inline contributingGuides/TS_STYLE.md:644]:
```suggestion
- [1.23](#ref-types) **Ref types**: Avoid using HTML elements while declaring refs. Please use React Native components where possible. React Native Web handles the references on its own. It also extends React Native components [with Interaction API](https://necolas.github.io/react-native-web/docs/interactions/) which should be used to handle Pointers and Mouse events. Exception of this rule is when we explicitly need to use functions available only in DOM and not in React Native, e.g. `getBoundingClientRect`. Then please declare ref type as `union` of React Native component and HTML element. When passing it to React Native component cast it as soon as possible using utility methods declared in `src/types/utils`
```

File: contributingGuides/TS_STYLE.md [hunk 3]
```diff
@@ -640,6 +641,21 @@ type Foo = {
...
+
+const ref = useRef<View | HTMLDivElement>();
+
+if (ref.current && 'getBoundingClientRect' in ref.current ){
+  ref.current.getBoundingClientRect();
+}
+
+<View ref={viewRef(ref)}>
+
+```
```
Comment (blazejkustra) [inline contributingGuides/TS_STYLE.md:657]: Let's add an example how it should be done usually with just `View` type, and the example above should be marked as an exception (we could split the above text into two to better visualize the meaning)

File: src/types/utils/viewForwardedRef.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import type {ForwardedRef} from 'react';
 import type {View} from 'react-native';
 
-const viewForwardedRef = (ref: ForwardedRef<View | HTMLDivElement>) => ref as ForwardedRef<View>;
+const viewForwardedRef = (ref: ForwardedRef<View | HTMLElement>) => ref as ForwardedRef<View>;
```
Comment (blazejkustra) [inline src/types/utils/viewForwardedRef.ts:4]: Why these changes in utils?

Comment (jnowakow) [reply src/types/utils/viewForwardedRef.ts:4]: `HTMLELement` is more generic then `HTMLDivElement` and this function is used to cast to `View` so I changed it to accept other HTML elements as well. It was more useful for the `textRef` where there can be more than one html element (`HTMLDivElement` or `HTMLFormElement` for example).

**Review by blazejkustra — APPROVED** [review]

File: contributingGuides/TS_STYLE.md [hunk 5]
```diff
@@ -640,6 +641,28 @@ type Foo = {
   export someVariable
   ```
 
+- [1.23](#ref-types) **Ref types**: Avoid using HTML elements while declaring refs. Please use React Native components where possible. React Native Web handles the references on its own. It also extends React Native components [with Interaction API](https://necolas.github.io/react-native-web/docs/interactions/) which should be used to handle Pointers and Mouse events. Exception of this rule is when we explicitly need to use functions available only in DOM and not in React Native, e.g. `getBoundingClientRect`. Then please declare ref type as `union` of React Native component and HTML element. When passing it to React Native component cast it as soon as possible using utility methods declared in `src/types/utils`
```
Comment (fabioh8010) [inline contributingGuides/TS_STYLE.md:644]:
```suggestion
- [1.23](#ref-types) **Ref types**: Avoid using HTML elements while declaring refs. Please use React Native components where possible. React Native Web handles the references on its own. It also extends React Native components with [Interaction API](https://necolas.github.io/react-native-web/docs/interactions/) which should be used to handle Pointer and Mouse events. Exception of this rule is when we explicitly need to use functions available only in DOM and not in React Native, e.g. `getBoundingClientRect`. Then please declare ref type as `union` of React Native component and HTML element. When passing it to React Native component assert it as soon as possible using utility methods declared in `src/types/utils`.
```

nits

File: contributingGuides/TS_STYLE.md [hunk 6]
```diff
@@ -640,6 +641,28 @@ type Foo = {
...
+
+Normal usage: 
+```ts
```
Comment (fabioh8010) [inline contributingGuides/TS_STYLE.md:647]:
```suggestion
```tsx
```

File: contributingGuides/TS_STYLE.md [hunk 7]
```diff
@@ -640,6 +641,28 @@ type Foo = {
...
+const ref = useRef<View>();
+
+<View ref={ref} onPointerDown={e => {#DO SOMETHING}}>
+```
+
+Exceptional usage where DOM methods are necessary:
+```ts
```
Comment (fabioh8010) [inline contributingGuides/TS_STYLE.md:654]:
```suggestion
```tsx
```

File: contributingGuides/TS_STYLE.md [hunk 8]
```diff
@@ -640,6 +641,28 @@ type Foo = {
...
+import viewRef from '@src/types/utils/viewRef';
+
+const ref = useRef<View | HTMLDivElement>();
+
+if (ref.current && 'getBoundingClientRect' in ref.current ){
```
Comment (fabioh8010) [inline contributingGuides/TS_STYLE.md:659]:
```suggestion
if (ref.current && 'getBoundingClientRect' in ref.current) {
```

File: contributingGuides/TS_STYLE.md [hunk 9]
```diff
@@ -640,6 +641,28 @@ type Foo = {
...
+<View ref={ref} onPointerDown={e => {#DO SOMETHING}}>
+```
+
+Exceptional usage where DOM methods are necessary:
+```tsx
+import viewRef from '@src/types/utils/viewRef';
+
+const ref = useRef<View | HTMLDivElement>();
+
+if (ref.current && 'getBoundingClientRect' in ref.current ) {
```
Comment (fabioh8010) [inline contributingGuides/TS_STYLE.md:659]:
```suggestion
if (ref.current && 'getBoundingClientRect' in ref.current) {
```

**Review by fabioh8010 — APPROVED** [review]
Last nit and we're good to open to review @jnowakow !

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by rushatgabhane — APPROVED** [review]
LGTM

**Review by marcochavezf — DISMISSED** [review]

**Comment by marcochavezf** [comment]
Conflicts

**Comment by marcochavezf** [comment]
@jnowakow friendly bump to resolve conflicts :)

**Comment by jnowakow** [comment]
@marcochavezf, sorry, I was out of office last week. I've resolved conflicts

**Review by marcochavezf — APPROVED** [review]
